src/coreclr/jit/emitxarch.cpp

```diff
@@ -1795,23 +1648,29 @@ unsigned emitter::emitOutputRexOrVexPrefixIfNeeded(instruction ins, BYTE* dst, c
         //   * X and B must be set                (0x61 validates bits 5-6)
         //   * m-mmmm must be 0-00001             (0x61 validates bits 0-4)
         //   * W must be unset                    (0x00 validates bit 7)
+
         if ((vexPrefix & 0xFFFF7F80) == 0x00C46100)
         {
-            // Encoding optimization calculation is not done while estimating the instruction
-            // size and thus over-predict instruction size by 1 byte.
-            // If there are IGs that will be aligned, do not optimize encoding so the
-            // estimated alignment sizes are accurate.
-            if (emitCurIG->igNum > emitLastAlignedIgNum)
+            // Encoding optimization calculation when estimating the instruction size is
+            // only done when optimizations are enabled since its more expensive. So when
+            // optimizations are disabled we'll have over-predicted the instruction size
+            // by one byte. Due to how IG alignment is done, we only want to emit the 2-byte
+            // prefix if optimizations are enabled or we know we won't negatively impact the
+            // estimated alignment sizes.
+
+            if (emitComp->opts.OptimizationEnabled() || (emitCurIG->igNum > emitLastAlignedIgNum))
```

Comment:
Talked with @kunalspathak and this is currently needed as we still try to do alignment when optimizations are disabled.

We may want to revisit that since OSR + TC should handle all the important cases and aligning debug code likely isn't worth the cycles required.